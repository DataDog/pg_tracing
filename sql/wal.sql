-- Generate queries with wal write
/*dddbs='postgres.db',traceparent='00-00000000000000000000000000000001-0000000000000001-01'*/ INSERT INTO pg_tracing_test VALUES(generate_series(1, 10), 'aaa');
/*dddbs='postgres.db',traceparent='00-00000000000000000000000000000002-0000000000000002-01'*/ UPDATE pg_tracing_test SET b = 'bbb' WHERE a = 7;
/*dddbs='postgres.db',traceparent='00-00000000000000000000000000000003-0000000000000003-01'*/ DELETE FROM pg_tracing_test WHERE a = 9;

-- Check WAL is generated for the above statements
SELECT trace_id, span_type, span_operation,
       wal_records > 0 as wal_records,
       wal_bytes > 0 as wal_bytes
FROM peek_ordered_spans;
CALL clean_spans();

-- Cleanup
CALL clean_spans();
