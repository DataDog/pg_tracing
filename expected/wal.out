-- Generate queries with wal write
/*dddbs='postgres.db',traceparent='00-00000000000000000000000000000001-0000000000000001-01'*/ INSERT INTO pg_tracing_test VALUES(generate_series(1, 10), 'aaa');
/*dddbs='postgres.db',traceparent='00-00000000000000000000000000000002-0000000000000002-01'*/ UPDATE pg_tracing_test SET b = 'bbb' WHERE a = 7;
/*dddbs='postgres.db',traceparent='00-00000000000000000000000000000003-0000000000000003-01'*/ DELETE FROM pg_tracing_test WHERE a = 9;
-- Check WAL is generated for the above statements
SELECT trace_id, span_type, span_operation,
       wal_records > 0 as wal_records,
       wal_bytes > 0 as wal_bytes
FROM peek_ordered_spans;
             trace_id             |  span_type   |                          span_operation                          | wal_records | wal_bytes 
----------------------------------+--------------+------------------------------------------------------------------+-------------+-----------
 00000000000000000000000000000001 | Insert query | INSERT INTO pg_tracing_test VALUES(generate_series($1, $2), $3); | t           | t
 00000000000000000000000000000001 | Planner      | Planner                                                          | f           | f
 00000000000000000000000000000001 | Executor     | ExecutorRun                                                      |             | 
 00000000000000000000000000000002 | Update query | UPDATE pg_tracing_test SET b = $1 WHERE a = $2;                  | t           | t
 00000000000000000000000000000002 | Planner      | Planner                                                          | f           | f
 00000000000000000000000000000002 | Executor     | ExecutorRun                                                      |             | 
 00000000000000000000000000000003 | Delete query | DELETE FROM pg_tracing_test WHERE a = $1;                        | t           | t
 00000000000000000000000000000003 | Planner      | Planner                                                          | f           | f
 00000000000000000000000000000003 | Executor     | ExecutorRun                                                      |             | 
(9 rows)

CALL clean_spans();
-- Cleanup
CALL clean_spans();
